### ISSUE TITLE
Fix logical bug in backend

### ISSUE THREAD

**Issue by TheFatBlue** [DFY-NCSU/CoinCanvas#12]
Some test cases in #11 will fail, indicating that the logic of backend has some flaw. Please fix them to pass all the test cases.

**Comment by TheFatBlue** [comment]
The failed cases are:
- `test_read_expenses_negative_limit`
- `test_read_expenses_negative_skip`
- `test_negative_amount`
- `test_empty_strings`
- `test_empty_password`
- `test_empty_full_name`

**Comment by TheFatBlue** [comment]
Here's detailed failure information:
```bash
================================================================================================= FAILURES ==================================================================================================
_____________________________________________________________________________________ TestRegister.test_empty_password ______________________________________________________________________________________

self = <tests.test_api_register.TestRegister object at 0x10410c2d0>, test_user = {'email': 'test1731875761.999737@example.com', 'full_name': 'name', 'password': 'pass'}

    def test_empty_password(self, test_user):
        """Test user creation with empty password"""
        invalid_user = test_user.copy()
        invalid_user["password"] = ""
        response = requests.post(f"{BASE_URL}/users/", json=invalid_user)
>       assert response.status_code == 422
E       assert 200 == 422
E        +  where 200 = <Response [200]>.status_code

test_api_register.py:95: AssertionError
_____________________________________________________________________________________ TestRegister.test_empty_full_name _____________________________________________________________________________________

self = <tests.test_api_register.TestRegister object at 0x10410c9d0>, test_user = {'email': 'test1731875762.2448@example.com', 'full_name': 'name', 'password': 'pass'}

    def test_empty_full_name(self, test_user):
        """Test user creation with empty full name"""
        invalid_user = test_user.copy()
        invalid_user["full_name"] = ""
        response = requests.post(f"{BASE_URL}/users/", json=invalid_user)
>       assert response.status_code == 422
E       assert 200 == 422
E        +  where 200 = <Response [200]>.status_code

test_api_register.py:102: AssertionError
__________________________________________________________________________________ TestCreateExpense.test_negative_amount ___________________________________________________________________________________

self = <tests.test_api_create_expenses.TestCreateExpense object at 0x107630290>
auth_headers = {'Authorization': 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJ0ZXN0X2V4cGVuc2VfMTczMTg3NTg2Mi4wNzM2MDdAZXhhbXBsZS5jb20iLCJleHAiOjE3MzE4Nzc2NjJ9.mjc7nopolCf-JR9ew2YjIZeUz2iVnfjeN1xn3I7ZkJQ', 'Content-Type': 'application/json'}
valid_expense = {'amount': 25.5, 'category': 'Food', 'description': 'Lunch at restaurant', 'payment_method': 'Credit Card'}

    def test_negative_amount(self, auth_headers, valid_expense):
        """Test expense creation with negative amount"""
        invalid_expense = valid_expense.copy()
        invalid_expense["amount"] = -50.00
        response = requests.post(
            f"{BASE_URL}/expenses/",
            json=invalid_expense,
            headers=auth_headers
        )
>       assert response.status_code == 422
E       assert 200 == 422
E        +  where 200 = <Response [200]>.status_code

test_api_create_expenses.py:140: AssertionError
___________________________________________________________________________________ TestCreateExpense.test_empty_strings ____________________________________________________________________________________

self = <tests.test_api_create_expenses.TestCreateExpense object at 0x1076314d0>
auth_headers = {'Authorization': 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJ0ZXN0X2V4cGVuc2VfMTczMTg3NTg2My4wMzQ4MjkxQGV4YW1wbGUuY29tIiwiZXhwIjoxNzMxODc3NjYzfQ.0eDqqFRllJp7GfRicoYQYdbmF8SASxi6lnGPEIwCwJI', 'Content-Type': 'application/json'}
valid_expense = {'amount': 25.5, 'category': 'Food', 'description': 'Lunch at restaurant', 'payment_method': 'Credit Card'}

    def test_empty_strings(self, auth_headers, valid_expense):
        """Test expense creation with empty strings"""
        invalid_expense = valid_expense.copy()
        invalid_expense["category"] = ""
        invalid_expense["payment_method"] = ""
        response = requests.post(
            f"{BASE_URL}/expenses/",
            json=invalid_expense,
            headers=auth_headers
        )
>       assert response.status_code == 422
E       assert 200 == 422
E        +  where 200 = <Response [200]>.status_code

test_api_create_expenses.py:162: AssertionError
_____________________________________________________________________________ TestReadExpense.test_read_expenses_negative_limit _____________________________________________________________________________

self = <tests.test_api_read_expenses.TestReadExpense object at 0x1063d6450>
auth_headers = {'Authorization': 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJ0ZXN0X2V4cGVuc2VfcmVhZF8xNzMxODc1OTA4LjYyNjk...FtcGxlLmNvbSIsImV4cCI6MTczMTg3NzcwOX0.0g2l9d-gdsie9POPxB2-st4j1F-j_h21Z-nYEdDuzL8', 'Content-Type': 'application/json'}

    def test_read_expenses_negative_limit(self, auth_headers):
        """Test reading expenses with negative limit"""
        response = requests.get(
            f"{BASE_URL}/expenses/?limit=-1",
            headers=auth_headers
        )
>       assert response.status_code == 422  # Validation error
E       assert 200 == 422
E        +  where 200 = <Response [200]>.status_code

test_api_read_expenses.py:146: AssertionError
_____________________________________________________________________________ TestReadExpense.test_read_expenses_negative_skip ______________________________________________________________________________

self = <tests.test_api_read_expenses.TestReadExpense object at 0x1063d6cd0>
auth_headers = {'Authorization': 'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJ0ZXN0X2V4cGVuc2VfcmVhZF8xNzMxODc1OTA5LjEwNjk...FtcGxlLmNvbSIsImV4cCI6MTczMTg3NzcwOX0.84caJQyrAAZmtjFEJPTIZm6RoOy-5RCGlvYxGT3mb4o', 'Content-Type': 'application/json'}

    def test_read_expenses_negative_skip(self, auth_headers):
        """Test reading expenses with negative skip"""
        response = requests.get(
            f"{BASE_URL}/expenses/?skip=-1",
            headers=auth_headers
        )
>       assert response.status_code == 422  # Validation error
E       assert 200 == 422
E        +  where 200 = <Response [200]>.status_code

test_api_read_expenses.py:154: AssertionError
```

**Comment by TomasAndersonFang** [comment]
Thanks for your reporting, I'll try to fix!

**Comment by TheFatBlue** [comment]
solved in #19
